### ISSUE TITLE
Docs: missing BackoffSupervisor.PropsWithSupervisorStrategy documentation

### ISSUE THREAD

**Issue by christallire** [akkadotnet/akka.net#5065]
**Please describe what you are trying to understand**
BackoffSupervisor.PropsWithSupervisorStrategy

**What did these pages not make clear?**
BackoffSupervisor.PropsWithSupervisorStrategy has no documentation or very basic documentation in XML doc, I'm not sure how it works

**How can we do it better?**
add more details

**Additional suggestions**
first I thought BackoffSupervisor.PropsWithSupervisorStrategy is the shorthand of BackoffSupervisor.Props(Backoff.OnFailure(...))
but it appears not. it behaves differently, could we have a more detailed explanation about it?

**Comment by ismaelhamed** [comment]
Not sure if there's much more that can be added to method description since it does exactly what is says: creates the props for an `BackoffSupervisor` actor with a custom supervision strategy, unlike `BackoffSupervisor.Props(Backoff.OnFailure(...))` and `BackoffSupervisor.Props(Backoff.OnStop(...))` that use the two built-in supervision strategies.

**Comment by christallire** [comment]
Yeah, this might be silly but hard to know as a newbie -- just wanted to share the experience.
